### DISCUSSION TITLE
What does data structure in PYG.TemporalData mean? And can it be in a data_list way?

### DISCUSSION THREAD

**Discussion by Kevin-shihello-world** [pyg-team/pytorch_geometric#5856] [Q&A]
As this question says. I'm trying to learn something from the temporal graph prediction. But the data structure in PYG.TemporalData is a bit confusing, I mean does this mean we can only use datasets with two nodes interacting at each time(t)? Can we in some ways make it happen to be in a data_list way? It may help a lot on tests like transportation graph prediction, we can't judge only two cars on a road at each time.
And the program I want to write has fixed node_num and change over time edge_num.
I would appreciate it if someone can help me find an answer. Thank you!

**Comment by rusty1s** [comment]
Thanks for your interest. You are correct that the current implementation of `TemporalData` at least requires uniform features across different types. I don't think the current implementation of a list of events requires only two nodes interacting at each time. If there are multiple events happening at the same time, they would exist as separate events with the same timestamp.

There also exists https://github.com/pyg-team/pytorch_geometric/issues/3230 for further discussions on how to improve temporal graph learning in PyG. Happy to get your input on this one :)
